Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,486 @@
+  filename = std::string(dir);
+  filename.append(m_OpenPMDFileType);
+
+  if (m_OneFilePerTS)
+    filename += amrex::Concatenate(prefix, ts);
+  else
+    filename.append(prefix);
+
+  filename += std::string(".") + m_OpenPMDFileType;
+}
+
+
+void WarpXOpenPMDParticle::SetOpenPMDType(const std::string& type)
+{
+  m_OpenPMDFileType = type;
+}
+
+void WarpXOpenPMDParticle::SetStep(int ts)
+{
+  if (ts < 0)
+    return;
+
+  m_CurrentStep =  ts;
+
+  std::string filename;
+  if (m_OneFilePerTS) {
+    GetFileName(filename, ts);
+    Init(filename, openPMD::AccessType::CREATE);
+  } else { // one file
+    if (nullptr == m_Series) {
+      GetFileName(filename, -1);
+      Init(filename, openPMD::AccessType::CREATE);
+    }
+  }
+}
+
+void
+WarpXOpenPMDParticle::Init(const std::string& filename,
+               openPMD::AccessType accessType)
+{
+  if (m_Series != nullptr) {
+    m_Series->flush();
+    delete m_Series;
+    m_Series = nullptr;
+  }
+
+  if (amrex::ParallelDescriptor::NProcs() > 1) {
+    m_Series = new openPMD::Series(filename,
+                   accessType,
+                   amrex::ParallelDescriptor::Communicator());
+    m_MPISize = amrex::ParallelDescriptor::NProcs();
+    m_MPIRank = amrex::ParallelDescriptor::MyProc();
+  }
+  else
+    m_Series = new openPMD::Series(filename, accessType);
+}
+
+
+void
+WarpXOpenPMDParticle::SaveContainerPlots(const std::unique_ptr<MultiParticleContainer>& mpc)
+{
+  std::vector<std::string> species_names = mpc->GetSpeciesNames();
+
+  for (unsigned i = 0, n = species_names.size(); i < n; ++i) {
+    //auto& pc = allcontainers[i];
+    //auto& pc =  mpc->GetParticleContainer(i);
+    auto& pc  = mpc->GetUniqueContainer(i);
+    if (pc->plot_species) {
+
+      amrex::Vector<std::string> real_names;
+      amrex::Vector<std::string> int_names;
+      amrex::Vector<int> int_flags;
+
+      real_names.push_back("weight");
+
+      real_names.push_back("momentum_x");
+      real_names.push_back("momentum_y");
+      real_names.push_back("momentum_z");
+
+      real_names.push_back("Ex");
+      real_names.push_back("Ey");
+      real_names.push_back("Ez");
+
+      real_names.push_back("Bx");
+      real_names.push_back("By");
+      real_names.push_back("Bz");
+
+#ifdef WARPX_DIM_RZ
+      real_names.push_back("theta");
+#endif
+      if(pc->do_field_ionization){
+    int_names.push_back("ionization_level");
+    // int_flags specifies, for each integer attribs, whether it is
+    // dumped to plotfiles. So far, ionization_level is the only
+    // integer attribs, and it is automatically dumped to plotfiles
+    // when ionization is on.
+    int_flags.resize(1, 1);
+      }
+
+      // Convert momentum to SI
+      pc->ConvertUnits(ConvertDirection::WarpX_to_SI);
+      // real_names contains a list of all particle attributes.
+      // pc->plot_flags is 1 or 0, whether quantity is dumped or not.
+
+      //
+      // NOTE: using 0 as iteration number b/c right now
```

Comment:
Sorry that comment was not valid anymore.  I will remove it.  
 Please ignore it. 

The file has all time steps in.  Otherwise format like plot00000.h5,  no underscore

Reply:
No problem, yes just feel free to push another update.